src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -508,9 +534,14 @@ const OfflineTokenRow = ({ onValueChanged, validationFailed }) => {
             setValidationState(validationStates.INPROGRESS);
             getAccessToken(offlineToken)
                     .then(out => {
+                        console.log(out);
                         const accessToken = out.trim();
                         onValueChanged("accessToken", accessToken);
                         setValidationState(validationStates.SUCCESS);
+
+                        // TODO encrypt offline token
```

Comment:
But with what? I don't want c-machines to become a password manager, and the user having to set and remember some random password here. Either we accept the risk of stealing local storage secrets from a browser (which should be manageable, as localStorage is per origin), or we don't, and then we should instead use sessionStorage and leave longer-term secrets management to the user.

My gut feeling is that localStorage is okay, with some fixes (see other threads).

Reply:
How about using some information about the server which the user is connected to as a secret? E.g. its UUID?

Reply:
You mean the /etc/machine-id? But that's not secret at all. We are not really concerned about offline attacks here -- people should encrypt their home directories. In a running browser, anyone can read any of the public OS and the user's files. The only secret which we have generally available is the user's login password (but that's not accessible at all from the browser) and the browser's general password keyring (but there's a like a million implementations of that, and websites can't generally control these).